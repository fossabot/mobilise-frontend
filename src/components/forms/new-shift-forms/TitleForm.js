import React from 'react';
import { Form } from 'react-bootstrap';
import { Typeahead } from 'react-bootstrap-typeahead';

export default class TitleForm extends React.Component {
  submit = () => console.log('Location and event type inputted'); // TODO create submit function

  render() {
    return (
      <Form id="new-shift-form-part-1" onSubmit={this.submit}>
        <Form.Group controlId="titleForm">
          <Form.Label>Shift Title</Form.Label>
          <Typeahead // TODO make async
            id="Title"
            placeholder="Choose an event Title, or create a new one"
            allowNew
            onChange={selected => console.log(selected)}
            options={['Cake Sale', 'Food Managing']} // TODO
          />
        </Form.Group>
        <Form.Group controlId="descriptionForm">
          <Form.Label>Description</Form.Label>
          <Form.Control as="textarea" rows="3" />
        </Form.Group>
      </Form>
    );
  }
}
