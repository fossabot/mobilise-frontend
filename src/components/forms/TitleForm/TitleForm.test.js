import React from 'react';
import { shallow } from 'enzyme';
import TitleForm from './TitleForm';

const testTitleProps = {
  title: '',
  shiftTitleOptions: ['Fundraiser', 'Regular'],
  handleChange: () => {}
};

let wrapper;

beforeEach(() => {
  wrapper = shallow(<TitleForm titleProps={testTitleProps} />);
});

describe('Title Form Component', () => {
  it('renders the form correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should contain a typeahead', () => {
    expect(wrapper.find('TypeaheadComponent')).toHaveLength(1);
  });
});
