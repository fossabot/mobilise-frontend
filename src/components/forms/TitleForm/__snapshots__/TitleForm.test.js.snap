// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Form Component renders the form correctly 1`] = `ShallowWrapper {}`;
